nixos/modules/tasks/network-interfaces-scripted.nix

```diff
@@ -183,14 +183,15 @@ let
                 ${flip concatMapStrings ips (ip:
                   let
                     cidr = "${ip.address}/${toString ip.prefixLength}";
+                    brd = "${ip.broadcast}";
                   in
                   ''
-                    echo "${cidr}" >> $state
+                    echo "${cidr} ${brd}" >> $state
                     echo -n "adding address ${cidr}... "
-                    if out=$(ip addr add "${cidr}" dev "${i.name}" 2>&1); then
+                    if out=$(ip addr add "${cidr}" brd "${brd}" dev "${i.name}" 2>&1); then
```

Comment:
Can you also add this feature to `network-interfaces-systemd.nix`? I assume that `+` and `-` does not work there so there should be an `assert` to catch this.

Reply:
It [appears](https://www.freedesktop.org/software/systemd/man/systemd.network.html#Broadcast=) that the default there is "+", and there's no equivalent to "-". That entry also reminded me that IPv6 doesn't have network-specific broadcast addresses, just the various scoped multicast addresses, so removing the option on IPv6 would be preferable.